### PR TITLE
[RA2 Ch2]: Correction of linter errors

### PR DESCRIPTION
**Pull request by CsatariGergely**
Fixes: #3007

### CODE REVIEW

**Review by ulikleber — APPROVED** [review]

File: doc/ref_arch/kubernetes/chapters/chapter02.rst [hunk 1]
```diff
@@ -4,8 +4,8 @@ Architecture Requirements
 Introduction
 ------------
 
-This chapter will use the requirements defined in the overall Reference Model and only make additional entries in section
-`2.3 <#2.3>`__ if there are additional requirements needed for this Reference Architecture.
+This chapter will use the requirements defined in the overall Reference Model and only make additional entries in
+section `2.3 <#2.3>`__ if there are additional requirements needed for this Reference Architecture.
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter02.rst:8]: can we not use section numbers as they may change?

section `Reference Model Requirements`__ if there are additional ...

File: doc/ref_arch/kubernetes/chapters/chapter02.rst [hunk 2]
```diff
@@ -860,7 +860,7 @@ Cloud Infrastructure Security Requirements
 |                                        |           | versions.                            |                         |
 +----------------------------------------+-----------+--------------------------------------+-------------------------+
 |`7.9.5`                                 |sec.img.008| Images **must not** include any      | `5.12 Secrets           |
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter02.rst:862]: Can we remove section numbers?

**Review by pgoyal01 — COMMENTED** [review]
Nearly every Table has an entry of the form: :doc:`ref_model/chapters/chapter04:Reference` where the "Reference" needs to be changed.

Section numbers should be removed from references.

Line 216 the reference has an error: " :doc:`ref_model/chapters/chapter04:4.2.6 Storage Ext` " it should be " :doc:`ref_model/chapters/chapter04:Storage Extensions` "

**Review by rgstori — APPROVED** [review]
approved pending reference fixes (or moving to a new issue)

**Comment by pgoyal01** [comment]
@CsatariGergely @rgstori There are at least two places where the reference link  is to "Reference" in a chapter -- no such section exists.

**Comment by rgstori** [comment]
many (but not all) of the changes here are in #3037, I would propose merging that first and then deconflicting manually

**Comment by rgstori** [comment]
many (but not all) of the changes here are in #3037, I would propose merging that first and then deconflicting manually

**Comment by collivier** [comment]
ok. I merge and fix all in a new PR

**Pull request merged**